modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/sqlquery/DbExecutor.scala

```diff
@@ -32,8 +34,8 @@ import scala.collection.immutable.IntMap
 /** Side-effecting ability to connect to database */
 trait DbExecutor[F[_]] {
 
-  /** Lookup a connection from mutable reference or initialize and put into the cache */
-  def getConnection(db: Rdbms, connRef: ConnectionRef[F])(implicit M: Monad[F]): F[Either[Throwable, Connection]]
+  /** Get a connection from the Hikari data source */
+  def getConnection(dataSource: HikariDataSource): F[Either[Throwable, Connection]]
```

Comment:
I can see we rely here mainly on `java.sql`, so maybe we could use `javax.sql.DataSource` here and pass hikari as implementation?